### PR TITLE
Verify clienter <-> interaction is happening

### PR DESCRIPTION
**Pull request by bveliqi**

### CODE REVIEW

File: adversarial_vision_challenge/server.py [hunk 1]
```diff
@@ -89,6 +90,7 @@ def _predict(image):
 
     @app.route("/")
     def main():  # pragma: no cover
+        cs_interaction_verifier.start()
```
Comment (spMohanty) [inline adversarial_vision_challenge/server.py]: @bveliqi : Shouldn't the verifier start right when it is initialized ? 
(I understand we might want to give a little bit of extra time in the beginning for the Attack to boot up in the beginning, but a convenient 3-5mins should be enough for all good purposes.)

Plus, allowing the attacker to start a new verifier on every GET call to `/` sounds a bit off to me.

Can we move the  `cs_interaction_verifier.start()` call to the `__init__` function of `InteractionVerifier` ?

Comment (spMohanty) [reply adversarial_vision_challenge/server.py]: Other than that, it works for me.
I am adding the necessary changes to the evaluator.

File: adversarial_vision_challenge/interaction_verifier.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                "Client has not sent any requests to to the server for more than {}s ".format(duration)
+            )
+
+
+class Caller:
+
+    def __init__(self):
+        # interval in seconds to check client <-> server interaction
+        self.__cs_interaction_verifier = InteractionVerifier()
+        self.__interval = int(os.getenv('CS_INTERACTION_TIMEOUT_INTERVAL', 5))
```
Comment (spMohanty) [inline adversarial_vision_challenge/interaction_verifier.py]: I have a feeling this variable naming will confuse us at some point of time. Maybe calling it `CS_INTERACTION_CHECK_INTERVAL` would make more sense ?

**Comment by spMohanty** [comment]
@bveliqi : You do not see any reviews here ?

File: adversarial_vision_challenge/server.py (hunk 1 above)
Comment (spMohanty) [reply adversarial_vision_challenge/server.py]: @bveliqi : Test comment.

**Comment by bveliqi** [comment]
@spMohanty not a single one.

**Review by bveliqi — COMMENTED** [review]
Test

**Comment by bveliqi** [comment]
I can see mine when I test it: https://github.com/bethgelab/adversarial-vision-challenge/pull/27#pullrequestreview-139032331

Not your though... are you sure you've clicked on 'Submit review' after commenting?

**Comment by bveliqi** [comment]
Not sure what you changed now, but now I received an eMail and I can see your comments: https://github.com/bethgelab/adversarial-vision-challenge/pull/27#pullrequestreview-138303275

File: adversarial_vision_challenge/server.py (hunk 1 above)
Comment (bveliqi) [reply adversarial_vision_challenge/server.py]: You're right. Fixed.

File: adversarial_vision_challenge/interaction_verifier.py (hunk 2 above)
Comment (bveliqi) [reply adversarial_vision_challenge/interaction_verifier.py]: You're right. Fixed.

**Comment by spMohanty** [comment]
Great !
LGTM (after the tests pass :D )

**Pull request merged**